src/App/Security/Authorization/Voter/PrivateDiscussionVoter.php

```diff
@@ -21,52 +21,33 @@ public function __construct(Doctrine $doctrine)
         $this->doctrine = $doctrine;
     }
 
-    public function supportsAttribute($attribute)
+    public function getSupportedAttributes()
     {
-        return in_array($attribute, array(
+        return array(
             self::VIEW,
             self::EDIT,
             self::ADD_MEMBER,
             self::UNREADER,
-        ));
+        );
     }
 
-    public function supportsClass($class)
+    public function getSupportedClasses()
     {
-        $supportedClass = 'App\Entity\PrivateDiscussion';
-
-        return $supportedClass === $class || is_subclass_of($class, $supportedClass);
+        return array('App\Entity\PrivateDiscussion');
     }
 
-    public function vote(TokenInterface $token, $discussion, array $attributes)
+    public function isGranted($attribute, $discussion, $user = null)
     {
-        // check if class of this object is supported by this voter
-        if (!$this->supportsClass(get_class($discussion))) {
-            return VoterInterface::ACCESS_ABSTAIN;
+        if (!$user instanceof UserInterface) {
+            return false;
         }
 
-        // check if the voter is used correctly, only allow one attribute
-        if (1 !== count($attributes)) {
-            throw new \InvalidArgumentException(
-                'Only one attribute is allowed'
+        if (!$user instanceof User) {
+            throw new \LogicException(
+                'The user is somehow not our User class!'
```

Comment:
Why not directly check for this User instance instead of UserInterface?
Moreover you have to check that `$user` is an object. `null instanceof Something` throws a fatal error. Same for `'anon.' instanceof Something` (If the user is anonymous, the firewall can return a string).

Reply:
The voters will be completely reviewed in a next PR since they are deprecated in the 2.8 version of Symfony